src/core/ext/transport/chttp2/transport/hpack_parser_table.h

```diff
@@ -30,6 +30,40 @@
 
 namespace grpc_core {
 
+using Memento = ParsedMetadata<grpc_metadata_batch>;
```

Comment:
This is probably too generic of a name to leave at the top level of the namespace.

Since we're having some build problems anyway, let's remove it and make `MementoRingBuffer` a child class of `HPackTable`.

Build problem:
```
var/local/git/grpc/src/core/ext/transport/chttp2/transport/hpack_parser_table.h:79:9: error: declaration of ‘using Memento = using Memento = class grpc_core::ParsedMetadata<grpc_metadata_batch>’ changes meaning of ‘Memento’ [-fpermissive]
   79 |   using Memento = Memento;
      |         ^~~~~~~
/var/local/git/grpc/src/core/ext/transport/chttp2/transport/hpack_parser_table.h:33:7: note: ‘Memento’ declared here as ‘using Memento = class grpc_core::ParsedMetadata<grpc_metadata_batch>’
   33 | using Memento = ParsedMetadata<grpc_metadata_batch>;
      |       ^~~~~~~
```

Reply:
Fixed